src/cargo/core/compiler/compilation.rs

```diff
@@ -384,6 +375,48 @@ fn fill_rustc_tool_env(mut cmd: ProcessBuilder, unit: &Unit) -> ProcessBuilder {
     cmd
 }
 
+fn get_sysroot_target_libdir(
+    bcx: &BuildContext<'_, '_>,
+) -> CargoResult<HashMap<CompileKind, PathBuf>> {
+    bcx.all_kinds
+        .iter()
+        .map(|&kind| {
+            Ok((
+                kind,
+                bcx.target_data
+                    .get_info(kind)
+                    .ok_or_else(|| {
+                        let target = match kind {
+                            CompileKind::Host => "host".to_owned(),
+                            CompileKind::Target(s) => s.short_name().to_owned(),
+                        };
+
+                        let dependency = bcx
+                            .unit_graph
+                            .iter()
+                            .find_map(|(k, _)| {
+                                if k.kind == kind {
+                                    let summary = k.pkg.manifest().summary();
+
+                                    Some(format!("{} v{}", summary.name(), summary.version()))
+                                } else {
+                                    None
+                                }
+                            })
```

Comment:
(nit: styling)

```suggestion
                            .find_map(|(u, _)| (unit.kind == kind).then_some(unit.pkg.summary().package_id()))
```

BTW, it might be overkill but by reusing `Display` impl for `PackageId` one can get more useful info when developing a local artifact dep. (And DRY rule applied 😆)


https://github.com/rust-lang/cargo/blob/b862c6df16bfe6f29da2237c7d1a86159dc85d19/src/cargo/core/package_id.rs#L210-L220